blocks/product-details/product-details.js

```diff
@@ -1,26 +1,270 @@
 /* eslint-disable import/no-extraneous-dependencies */
-import { events } from '@dropins/tools/event-bus.js';
-import { initializers } from '@dropins/tools/initializer.js';
+
 import {
   InLineAlert,
   Icon,
+  Button,
   provider as UI,
 } from '@dropins/tools/components.js';
-import * as productApi from '@dropins/storefront-pdp/api.js';
-import { render as productRenderer } from '@dropins/storefront-pdp/render.js';
+import { events } from '@dropins/tools/event-bus.js';
+import { initializers } from '@dropins/tools/initializer.js';
+import * as PDP from '@dropins/storefront-pdp/api.js';
+import { render as PDPProvider } from '@dropins/storefront-pdp/render.js';
 import { addProductsToCart } from '@dropins/storefront-cart/api.js';
-import ProductDetails from '@dropins/storefront-pdp/containers/ProductDetails.js';
+
+// Containers
+import ProductHeader from '@dropins/storefront-pdp/containers/ProductHeader.js';
+import ProductPrice from '@dropins/storefront-pdp/containers/ProductPrice.js';
+import ProductShortDescription from '@dropins/storefront-pdp/containers/ProductShortDescription.js';
+import ProductOptions from '@dropins/storefront-pdp/containers/ProductOptions.js';
+import ProductQuantity from '@dropins/storefront-pdp/containers/ProductQuantity.js';
+import ProductDescription from '@dropins/storefront-pdp/containers/ProductDescription.js';
+import ProductAttributes from '@dropins/storefront-pdp/containers/ProductAttributes.js';
+import ProductGallery from '@dropins/storefront-pdp/containers/ProductGallery.js';
 
 // Libs
 import {
-  getProduct,
-  getSkuFromUrl,
   setJsonLd,
   loadErrorPage, performCatalogServiceQuery, variantsQuery,
+  getSkuFromUrl,
+  getOptionsUIDsFromUrl,
 } from '../../scripts/commerce.js';
-import { getConfigValue } from '../../scripts/configs.js';
 import { fetchPlaceholders } from '../../scripts/aem.js';
+import { getConfigValue } from '../../scripts/configs.js';
+
+export default async function decorate(block) {
+  // Set Fetch Endpoint (Service)
+  PDP.setEndpoint(await getConfigValue('commerce-endpoint'));
+
+  // Set Fetch Headers (Service)
+  PDP.setFetchGraphQlHeaders({
+    'Content-Type': 'application/json',
+    'Magento-Environment-Id': await getConfigValue('commerce-environment-id'),
+    'Magento-Website-Code': await getConfigValue('commerce-website-code'),
+    'Magento-Store-View-Code': await getConfigValue('commerce-store-view-code'),
+    'Magento-Store-Code': await getConfigValue('commerce-store-code'),
+    'Magento-Customer-Group': await getConfigValue('commerce-customer-group'),
+    'x-api-key': await getConfigValue('commerce-x-api-key'),
+  });
+
+  // pre-fetch PDP data
+  const sku = getSkuFromUrl();
+  const optionsUIDs = getOptionsUIDsFromUrl();
+
+  const [product, labels] = await Promise.all([
+    PDP.fetchPDPData(sku, { optionsUIDs }),
+    fetchPlaceholders(),
+  ]);
+
+  // Initialize
+  initializers.register(PDP.initialize, {
+    sku,
```

Comment:
So is the SKU now mandatory in order to render the PDP container?

Reply:
If you are using the new containers, you must give an SKU to the initializer. However, this is optional if you use `ProductDetails` to make it backward compatible with integrations like Bulk. 

You can see examples of these two working on the PDP's sandbox:
- Legacy: https://git.corp.adobe.com/magento-datalake/storefront-pdp/blob/dca0cd5643693293951a6678851c5861c3432b46/examples/html-host/index.html#L224
- New: https://git.corp.adobe.com/magento-datalake/storefront-pdp/blob/dca0cd5643693293951a6678851c5861c3432b46/examples/html-host/index2.html#L203